### PR TITLE
Ignore didChangeWatchedFiles for open documents

### PR DESCRIPTION
**Pull request by dhuebner**

### CODE REVIEW

File: org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/LanguageServerImpl.xtend [hunk 1]
```diff
@@ -277,15 +278,20 @@ import org.eclipse.lsp4j.InitializedParams
...
 				} else {
-					dirtyFiles += toUri(fileEvent.uri)
+					dirtyFiles += key
 				}
-			}
-			workspaceManager.didChangeFiles(dirtyFiles, deletedFiles)
-		], [cancelIndicator , buildable | 
+			]
+			var Buildable buildable = []
+			if (!deletedFiles.isEmpty || !dirtyFiles.isEmpty)
```
Comment (cdietrich) [inline org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/LanguageServerImpl.xtend]: switching order would be consistent

**Comment by cdietrich** [comment]
@dhuebner could you test that with vscode.
see 

org.eclipse.xtext.ide.tests/src/org/eclipse/xtext/ide/tests/server/ServerTest.xtend / testIncrementalDeletion

could we add a testcase there with open vs non open

**Comment by dhuebner** [comment]
@cdietrich 

> could you test that with vscode.

I don't know how to test something with vscode. Is there a description somewhere? 

> could we add a testcase there with open vs non open

Will look in to it. Hopefully there is api to simply open an editor

**Comment by cdietrich** [comment]
(1) have a sample dsl with file (a) references files (b)
(2) have file (a) and (b) open
(3) delete (b) on disk - no errors
(4) close editor for (b) then (a) becomes red

**Comment by dhuebner** [comment]
@cdietrich I have a theia app, would it also work as a test platform?

**Comment by cdietrich** [comment]
what i dont know: how is this stuff implemented in different platforms.
what i want to know: do we make sure no stale files stay in index caused by this change.
we had problems with that before and i dont want it again

**Comment by dhuebner** [comment]
> what i dont know: how is this stuff implemented in different platforms.

They all should follow the protocol. I will write a junit tests as described in 
https://github.com/eclipse/xtext-core/pull/895#issuecomment-428083793 and than test the PR in a theia app. That should be enough

> what i want to know: do we make sure no stale files stay in index caused by this change.

There are tests for that I think. org.eclipse.xtext.ide.tests.server.OpenDocumentTest . The new test with open/close/build will additionally cover the case.

> we had problems with that before and i don't want it again

If the problem was solved there should be tests for that, if not we add them.

File: org.eclipse.xtext.ide.tests/src/org/eclipse/xtext/ide/tests/server/ServerTest.xtend [hunk 2]
```diff
@@ -100,7 +100,34 @@ class ServerTest extends AbstractTestLangLanguageServerTest {
...
+		assertTrue("Bar.testlang references type Foo from Foo.testlang: expect no error",
+			diagnostics.get(referencingFileURI).empty)
+
+		fileURI.deleteFile
+		languageServer.getWorkspaceService.didChangeWatchedFiles(
+			new DidChangeWatchedFilesParams(#[new FileEvent(fileURI, FileChangeType.Deleted)])
+		)
+		assertTrue("delete file on disk: expect no error", diagnostics.get(referencingFileURI).empty)
+		close(fileURI);
+		assertFalse("close deleted file: expect error", diagnostics.get(referencingFileURI).empty)
```
Comment (cdietrich) [inline org.eclipse.xtext.ide.tests/src/org/eclipse/xtext/ide/tests/server/ServerTest.xtend]: :+1:

Comment (dhuebner) [reply org.eclipse.xtext.ide.tests/src/org/eclipse/xtext/ide/tests/server/ServerTest.xtend]: Build is passing now. should I merge?

**Pull request merged**